<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsMeshRendererVectorSettingsWidgetBase</class>
 <widget class="QWidget" name="QgsMeshRendererVectorSettingsWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>332</width>
    <height>811</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Displaying Method</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QComboBox" name="mDisplayingVectorMethodcomboBox">
        <item>
         <property name="text">
          <string>Arrows</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Streamlines</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Particle traces</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="mStreamLinesSeedingMethodGroupBox">
     <property name="title">
      <string>Streamlines Seeding Method</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="3,2">
      <item>
       <widget class="QComboBox" name="mStreamlinesSeedingMethodComboBox">
        <item>
         <property name="text">
          <string>Regular grid</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Regular on border</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Randomly on mesh</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Density</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="mStreamlinesDensitySpinBox">
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>5.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="mStreamlinesWidthColorGroupBox">
     <property name="title">
      <string>Streamlines Width and Color</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Line Width</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="mStreamlinesWidthSpinBox">
          <property name="singleStep">
           <double>0.200000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QRadioButton" name="mStreamlinesFixedColorRadioButton">
          <property name="text">
           <string>Fixe color</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="mStreamlinesColorRampRadioButton">
          <property name="text">
           <string>Color ramp</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QgsColorButton" name="mStreamlinesFixeColorButton">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QgsColorRampShaderWidget" name="mStreamlinesColorRampShaderWidget" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="arrowWidthColorGroupBox">
     <property name="title">
      <string>Line Width and Color</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QgsDoubleSpinBox" name="mLineWidthSpinBox"/>
      </item>
      <item>
       <widget class="QgsColorButton" name="mColorWidget"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="filterByMagGroupBox">
     <property name="title">
      <string>Filter by Magnitude</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="minimumMagLabel">
        <property name="text">
         <string>Min</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="mMinMagLineEdit"/>
      </item>
      <item>
       <widget class="QLabel" name="maximumMagLabel">
        <property name="text">
         <string>Max</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="mMaxMagLineEdit"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="mDisplayVectorsOnGridGroupBox">
     <property name="title">
      <string>Display Vectors on User Grid</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="xSpacingLabel">
          <property name="text">
           <string>X Spacing</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="ySpacingLabel">
          <property name="text">
           <string>Y Spacing</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QgsSpinBox" name="mXSpacingSpinBox">
          <property name="suffix">
           <string> px</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>8000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="mYSpacingSpinBox">
          <property name="suffix">
           <string> px</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>5000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="headOptionsGroupBox">
     <property name="title">
      <string>Head Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="headWidthLabel">
        <property name="text">
         <string>Width</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="mHeadWidthLineEdit"/>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="percShaftLenLabel">
        <property name="text">
         <string>% of Shaft Length</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="headLengthLabel">
        <property name="text">
         <string>Length</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="mHeadLengthLineEdit"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="percShaftLenLabel_2">
        <property name="text">
         <string>% of Shaft Length</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="generalOptionsGroupBox">
     <property name="title">
      <string>Arrow Length</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="mShaftLengthComboBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QComboBox::AdjustToContents</enum>
        </property>
        <item>
         <property name="text">
          <string>Defined by Min and Max</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Scaled to Magnitude</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Fixed</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="mShaftOptionsStackedWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="minimumShaftLabel">
            <property name="text">
             <string>Minimum</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mMinimumShaftLineEdit"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="maximumShaftLabel">
            <property name="text">
             <string>Maximum</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="mMaximumShaftLineEdit"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="scaleByFactorOfLabel">
            <property name="text">
             <string>Scale by a Factor of</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mScaleShaftByFactorOfLineEdit"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3">
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="LengthLabel">
            <property name="text">
             <string>Length</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mShaftLengthLineEdit"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsColorButton</class>
   <extends>QPushButton</extends>
   <header>qgscolorbutton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsColorRampShaderWidget</class>
   <extends>QWidget</extends>
   <header>raster/qgscolorrampshaderwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
